### DISCUSSION TITLE
Easier way to watch test + files for changes?

### DISCUSSION THREAD

**Discussion by danawoodman** [vlang/v#19343] [Questions and Answers] [answered]
Is there a way to run all tests in a project and re-run when a file changes that is easier than this?

```sh
v watch -c test -a src/lexer/lexer.v src/lexer/lexer_test.v
```

I've tried a few combos of the flags but this is the only one I can get working currently.

**Comment by spytheman** [comment, marked as answer]
> Is there a way to run all tests in a project and re-run when a file changes?

No, `v watch` monitors the files used by a single program currently, while `v test` runs many (all found `_test.v` files), and currently can not report back to `v watch`  what files were needed for all found test files.

If your project has a single `_test.v` file however, it is a single program, even if it has many test functions in it, and you can run it without `v test` with:
`v watch -c src/lexer/lexer_test.v` .

It is a good idea for a future `v test` enhancement, so that `v watch -c test .` could work, without more flags needed.

**Reply by danawoodman** [reply]
Thanks for the detailed reply!

Would love to see `v watch -c test .` as it would make testing your whole project simple
